sdlib/d/gc/region.d

```diff
@@ -352,13 +352,37 @@ unittest enormous {
 }
 
 struct Region {
-	void* address;
+	/**
+	 * This is a bitfield containing the following elements:
+	 *  - c: The alloc class.
+	 *  - a: The address (BlockSize-aligned)
+	 *  - g: The generation.
+	 * 
+	 * 63    56 55    48 47    40 39    32 31    24 23    16 15     8 7      0
+	 * ........ cccccccc aaaaaaaa aaaaaaaa aaaaaaaa aaa..... ........ gggggggg
+	 */
+	ulong bits;
+
+	// Verify our assumptions.
+	static assert(LgAddressSpace <= 48, "Address space too large!");
+
+	enum SignificantAddressBits = LgAddressSpace - LgBlockSize;
+	static assert(SignificantAddressBits <= 32,
+	              "Unable to pack address in bits!");
```

Comment:
I don't think this is asserting anything we care about.

Reply:
On the other hand, we don't check that we have enough in the low bits, which is an assumption we make.